rtengine/dcraw.cc

```diff
@@ -6292,11 +6292,11 @@ void CLASS parse_mos (int offset)
 void CLASS linear_table (unsigned len)
 {
   int i;
-  if (len > 0x1000) len = 0x1000;
+  if (len > 0x10000) len = 0x10000;
```

Comment:
If you change this conditional, wouldn't there be a problem for the intermediate range? I.e. when len > 0x1000 and len < 0x10000?

Reply:
All entries above the tag provided number of entries will all have the last entry value and maximum will also have it.

With this patch for 4096 entries all the entries from index 4096 to 65535 will have the value of curve[4095].

In the current code, instead, all entries above 4096 will instead have value of 0 since they aren't populated (or are they populated in another code path?).

So there's something different but I'm unsure if this will have bad effects.